packages/test/test-end-to-end-tests/src/test/compression.spec.ts

```diff
@@ -85,13 +96,65 @@ const compressionSuite = (getProvider) => {
 			if (provider.type === "TestObjectProviderWithVersionedLoad") {
 				this.skip();
 			}
+			await setupContainers();
 			const value = generateRandomStringOfSize(5);
 			localMap.set("testKey", value);
 
 			await provider.ensureSynchronized();
 			assert.strictEqual(localMap.get("testKey"), value);
 			assert.strictEqual(remoteMap.get("testKey"), value);
 		});
+
+		const messageGenerationOptions = generatePairwiseOptions<{
+			compressionAndChunking?: {
+				chunking: boolean;
+			};
+			grouping: boolean;
+		}>({
+			compressionAndChunking: [undefined, { chunking: false }, { chunking: true }],
+			grouping: [true, false],
+		});
+
+		messageGenerationOptions.forEach((option) => {
```

Comment:
I'm curious if this generates all 6 combos or not.  For larger sets of options, it only makes sure each option shows up in at least one pair.

Reply:
It generates all 6 combos:

```
    ✔ Correctly processes incoming messages: compression [false] chunking [false] grouping [true]
    ✔ Correctly processes incoming messages: compression [false] chunking [false] grouping [false]
    ✔ Correctly processes incoming messages: compression [true] chunking [false] grouping [true]
    ✔ Correctly processes incoming messages: compression [true] chunking [false] grouping [false]
    ✔ Correctly processes incoming messages: compression [true] chunking [true] grouping [true]
    ✔ Correctly processes incoming messages: compression [true] chunking [true] grouping [false]
```